File: scripts/make-bg-transparent.js
#!/usr/bin/env node

const sharp = require('sharp');
const path = require('path');
const fs = require('fs');

const mediaDir = path.join(__dirname, '../dist/media');

async function processImages() {
  // PNGファイルの一覧を取得
  const pngFiles = fs.readdirSync(mediaDir)
    .filter(file => file.endsWith('.png'))
    .map(file => path.join(mediaDir, file));

  console.log(`Found ${pngFiles.length} PNG files. Processing...`);

  for (const filePath of pngFiles) {
    const fileName = path.basename(filePath);
    try {
      // 画像を読み込み、ピクセル処理で白背景を透明化
      const { data, info } = await sharp(filePath)
        .ensureAlpha()
        .raw()
        .toBuffer({ resolveWithObject: true });

      const pixelCount = data.length / 4;  // RGBA = 4 channels

      // ピクセル処理：白（255,255,255）または明るい色をアルファ0（透明）に
      for (let i = 0; i < pixelCount; i++) {
        const idx = i * 4;
        const r = data[idx];
        const g = data[idx + 1];
        const b = data[idx + 2];

        // 白または非常に明るい色（R,G,Bがすべて250以上）
        if (r >= 250 && g >= 250 && b >= 250) {
          data[idx + 3] = 0;  // アルファ値を0（完全透明）に
        }
      }

      // 処理済み画像を保存
      await sharp(data, {
        raw: {
          width: info.width,
          height: info.height,
          channels: 4
        }
      })
      .png()
      .toFile(filePath);

      console.log(`✓ ${fileName}`);
    } catch (error) {
      console.error(`✗ ${fileName}: ${error.message}`);
    }
  }

  console.log('Done');
}

processImages().catch(error => {
  console.error('Fatal error:', error);
  process.exit(1);
});

File: scripts/make_bg_transparent.py
#!/usr/bin/env python3

from PIL import Image
import os
from pathlib import Path

def make_white_transparent(image_path, brightness_threshold=220):
    """
    白または白っぽい色をアルファ値0（透明）に変換
    brightness_threshold: 明るさの閾値（0-255）
    """
    img = Image.open(image_path)
    
    # RGBAに変換（アルファチャネルを追加）
    img = img.convert('RGBA')
    
    # ピクセルデータを取得
    data = img.getdata()
    
    # 新しいピクセルデータを作成
    new_data = []
    for item in data:
        r, g, b = item[:3]
        a = item[3] if len(item) > 3 else 255
        
        # RGB値の平均（明るさ）を計算
        brightness = (int(r) + int(g) + int(b)) // 3
        
        # 白っぽい色（明るさがthreshold以上かつRGBがほぼ同じ値）
        # アンチエイリアス線を含める
        if brightness >= brightness_threshold:
            # さらに、彩度が低い（グレースケール的）かどうかで判定
            # R,G,Bの最大値と最小値の差が小さい = 彩度が低い
            max_val = max(r, g, b)
            min_val = min(r, g, b)
            saturation = max_val - min_val
            
            # 彩度が30以下で明るい = 白っぽい
            if saturation <= 30:
                a = 0  # 透明に
        
        new_data.append((r, g, b, a))
    
    # 新しいピクセルデータを設定
    img.putdata(new_data)
    
    # PNGで保存
    img.save(image_path, 'PNG')
    return True

def main():
    media_dir = Path(__file__).parent.parent / 'dist' / 'media'
    
    if not media_dir.exists():
        print(f"Error: {media_dir} directory not found")
        return
    
    # PNGファイルを検索
    png_files = sorted(media_dir.glob('*.png'))
    
    if not png_files:
        print(f"No PNG files found in {media_dir}")
        return
    
    print(f"Found {len(png_files)} PNG files. Processing...")
    
    for png_file in png_files:
        try:
            make_white_transparent(str(png_file))
            print(f"✓ {png_file.name}")
        except Exception as e:
            print(f"✗ {png_file.name}: {e}")
    
    print("Done")

if __name__ == '__main__':
    main()
